src/OpenLoco/src/Graphics/SoftwareDrawingContext.cpp

```diff
@@ -1453,14 +1453,251 @@ namespace OpenLoco::Gfx
+                    }
+
+                    case ControlCodes::newlineXY:
+                    {
+                        uint8_t offset = *str;
+                        str++;
+                        pos.x = loc.x + offset;
+
+                        offset = *str;
+                        str++;
+                        pos.y = loc.y + offset;
+
+                        break;
+                    }
+
+                    case ControlCodes::Font::small:
+                        setCurrentFontSpriteBase(Font::small);
+                        break;
+                    case ControlCodes::Font::large:
+                        setCurrentFontSpriteBase(Font::large);
+                        break;
+                    case ControlCodes::Font::regular:
+                        setCurrentFontSpriteBase(Font::medium_normal);
+                        break;
+                    case ControlCodes::Font::bold:
+                        setCurrentFontSpriteBase(Font::medium_bold);
+                        break;
+                    case ControlCodes::Font::outline:
+                        _currentFontFlags = _currentFontFlags | TextDrawFlags::outline;
+                        break;
+                    case ControlCodes::Font::outlineOff:
+                        _currentFontFlags = _currentFontFlags & ~TextDrawFlags::outline;
+                        break;
+                    case ControlCodes::windowColour1:
+                    {
+                        auto hue = _windowColours[0].c();
+                        setTextColours(Colours::getShade(hue, 7), PaletteIndex::index_0A, PaletteIndex::index_0A);
+                        break;
+                    }
+                    case ControlCodes::windowColour2:
+                    {
+                        auto hue = _windowColours[1].c();
+                        setTextColours(Colours::getShade(hue, 9), PaletteIndex::index_0A, PaletteIndex::index_0A);
+                        break;
+                    }
+                    case ControlCodes::windowColour3:
+                    {
+                        auto hue = _windowColours[2].c();
+                        setTextColours(Colours::getShade(hue, 9), PaletteIndex::index_0A, PaletteIndex::index_0A);
+                        break;
+                    }
+                    case ControlCodes::windowColour4:
+                    {
+                        auto hue = _windowColours[3].c();
+                        setTextColours(Colours::getShade(hue, 9), PaletteIndex::index_0A, PaletteIndex::index_0A);
+                        break;
+                    }
+
+                    case ControlCodes::inlineSpriteStr:
+                    {
+                        str += 4;
+                        // Not handled for YOffsets code
+                        assert(false);
+                        break;
+                    }
+
+                    case ControlCodes::Colour::black:
+                        setTextColour(0);
```

Comment:
Isn't this just `Colour_t`?

Reply:
The vast majority of this code is lifted from drawString but yes should probably make this a table or something.

Reply:
checked and no its only a limited subset of colour.h.

Reply:
Ah, I see only the three first entries match, indeed. Perhaps we should introduce a separate TextColour enum, then?